src/test/java/org/broadinstitute/hellbender/IntelInflaterIntegrationTest.java

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.hellbender;
+
+import com.intel.gkl.compression.IntelInflater;
+import org.broadinstitute.hellbender.utils.NativeUtils;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.test.SamAssertionUtils;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Test that it's possible to load libIntelInflater
+ */
+public class IntelInflaterIntegrationTest extends CommandLineProgramTest {
+
+    private static final String INPUT_DIR = publicTestDir + "/large";
+    private static final String INPUT_FILE = "CEUTrio.HiSeq.WGS.b37.NA12878.20.21.tiny.md.bam";
+
+    private boolean isIntelInflaterSupported() {
+        return (NativeUtils.runningOnLinux() || NativeUtils.runningOnMac()) && !NativeUtils.runningOnPPCArchitecture();
+    }
+
+    @Test
+    public void testIntelInflaterIsAvailable(){
+        if ( ! NativeUtils.runningOnLinux()  && ! NativeUtils.runningOnMac()) {
+            throw new SkipException("IntelInflater not available on this platform");
+        }
+
+        if ( NativeUtils.runningOnPPCArchitecture() ) {
+            throw new SkipException("IntelInflater not available for this architecture");
+        }
+
+        Assert.assertTrue(new IntelInflater().load(null), "IntelInflater shared library was not loaded. " +
+                "This could be due to a configuration error, or your system might not support it.");
+    }
+
+    @Test
+    public void testIntelInflaterDeflaterWithPrintReads() throws Exception {
+        if (!isIntelInflaterSupported()) {
```

Comment:
Also, can you add two additional test cases: JDK inflater + Intel Deflater, and Intel inflater + JDK deflater, using the same bam? Probably makes sense to add a `DataProvider` that passes in booleans for `--use_jdk_inflater` and `--use_jdk_deflater`.